lib/src/middleware/reselect_after_props.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:built_collection/built_collection.dart';
+import 'package:redux/redux.dart';
+import 'package:scadnano/src/actions/actions.dart' as actions;
+import 'package:scadnano/src/state/app_state.dart';
+import 'package:scadnano/src/state/design.dart';
+import 'package:scadnano/src/state/strand.dart';
+
+bool are_batch_actions(dynamic action) {
+  for (var action in action.actions) {
+    if (!(action is actions.StrandNameSet ||
+        action is actions.StrandOrSubstrandColorSet ||
+        action is actions.ScaffoldSet ||
+        action is actions.SubstrandNameSet ||
+        action is actions.StrandLabelSet ||
+        action is actions.SubstrandLabelSet ||
+        action is actions.RemoveDNA)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+reselect_after_props(Store<AppState> store, action, NextDispatcher next) {
```

Comment:
I think a more descriptive name is `reselected_strands_after_props_updated`. Also, did you mean to leave this PR as a draft?